opensquirrel/passes/router/astar_router.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+import math
+
+import networkx as nx
+
+from opensquirrel.exceptions import NoRoutingPathError
+from opensquirrel.ir import IR, SWAP, Gate
+from opensquirrel.passes.router import Router
+from opensquirrel.passes.router.heuristics import DISTANCE_FUNCTIONS, DistanceMetric
```

Comment:
To be adjusted when using the `calculate_distance` function, iso the `DISTANCE_FUNCTIONS` dict.

```suggestion
from opensquirrel.passes.router.heuristics import DistanceMetric, calculate_distance
```